qiskit/pulse/schedule.py

```diff
@@ -517,7 +519,7 @@ def _add_timeslots(self,
         Raises:
             PulseError: If timeslots overlap or an invalid start time is provided.
         """
-        if not isinstance(time, int):
+        if not isinstance(time, (int, np.integer)):
```

Comment:
This may be more general as detailed here https://stackoverflow.com/questions/37726830/how-to-determine-if-a-number-is-any-type-of-int-core-or-numpy-signed-or-not

```suggestion
        if not np.issubdtype(type(time), np.integer):
```

Reply:
Ack, I'd found that function but not the full explanation of differences between the two. Will push changes.